packages/main/src/Popover.js

```diff
@@ -388,21 +393,25 @@ class Popover extends Popup {
 			width = rect.width;
 			height = rect.height;
 
-			return { width, height };
+			return Promise.resolve({ width, height });
 		}
 
 		this.style.visibility = "hidden";
 		this.style.display = "block";
 
-		rect = this.getBoundingClientRect();
+		return new Promise(resolve => {
```

Comment:
can you make this an async arrow fn? IMO It will get more readable.

* up to you

Reply:
done

Reply:
actually that is not possible, because requestAnimationFrame works with a callback and not with a promise